sharding/peers.go

```diff
@@ -33,6 +35,54 @@ type peer struct {
 	address string
 }
 
+func (p *Peers) SmallestMin() (string, error) {
+	if len(p.peers) == 0 {
```

Comment:
Likewise, don't think you need this, sine for an empty list you'll return `prevNum+1`, and `prevNum` would be 0.